### PR TITLE
Parametrize conda installer

### PR DESCRIPTION
**Pull request by astaric**
##### Issue
Conda installer creation script had some values parametrized, while others were hardcoded throughout the file.

##### Description of changes
Use existing parameters wherever possible, add new parameters where needed.

### CODE REVIEW

File: scripts/windows/orange-conda.nsi [hunk 1]
```diff
@@ -702,19 +716,19 @@ Function un.Register
...
+               "${WINDOWS_UNINSTALL_REGKEY}\${INSTALL_REGISTRY_KEY}" InstallLocation
     ${If} $un.InstallDir != ""
     ${AndIf} $un.InstallDir == $InstDir
-        ${LogWrite} "Deleting reg key: ${WINDOWS_UNINSTALL_REGKEY}\${APPNAME}"
-        DeleteRegKey SHCTX "${WINDOWS_UNINSTALL_REGKEY}\${APPNAME}"
+        ${LogWrite} "Deleting reg key: ${WINDOWS_UNINSTALL_REGKEY}\${INSTALL_REGISTRY_KEY}"
+        DeleteRegKey SHCTX "${WINDOWS_UNINSTALL_REGKEY}\${INSTALL_REGISTRY_KEY}"
     ${Else}
         ${LogWrite} "InstallLocation from \
                      ${WINDOWS_UNINSTALL_REGKEY}\${APPNAME} does not match \
```
Comment (ales-erjavec) [inline scripts/windows/orange-conda.nsi]: Also change the ${APPNAME} -> ${INSTALL_REGISTRY_KEY} here.

**Comment by astaric** [comment]
Fun fact, we already have
```
# Full key path for the application uninstall entry in Add/Remove Programs
!define APPLICATION_UNINSTALL_REGKEY "${WINDOWS_UNINSTALL_REGKEY}\${INSTALL_REGISTRY_KEY}"
```

(though it was not used anywhere). I'll use it where applicable.

File: scripts/windows/orange-conda.nsi [hunk 2]
```diff
@@ -644,51 +658,51 @@ Section -Register SectionRegister
...
     WriteRegStr SHELL_CONTEXT \
-        "Software\Classes\OrangeCanvas\DefaultIcon" "" \
-        "$PythonPrefix\share\orange3\icons\OrangeOWS.ico"
+        "Software\Classes\${INSTALL_REGISTRY_KEY}\DefaultIcon" "" \
+        "$PythonPrefix\share\${ICONDIR}\OrangeOWS.ico"
     WriteRegStr SHELL_CONTEXT \
-        "Software\Classes\OrangeCanvas\Shell\Open\Command\" "" \
-        '"$PythonExecPrefix\pythonw.exe" -m Orange.canvas "%1"'
+        "Software\Classes\${INSTALL_REGISTRY_KEY}\Shell\Open\Command\" "" \
+        '"$PythonExecPrefix\pythonw.exe" -m "{LAUNCHERMODULE} "%1"'
```
Comment (ales-erjavec) [inline scripts/windows/orange-conda.nsi]: This `-m "{LAUNCHERMODULE} "%1"` should be `-m ${LAUNCHERMODULE} "%1"`

Comment (astaric) [reply scripts/windows/orange-conda.nsi]: Thanks. I have unified this line to match others as well ("" around -m flag). Is that ok or should it be avoided in some cases?

File: scripts/windows/orange-conda.nsi [hunk 3]
```diff
@@ -561,7 +561,7 @@ Section -Launchers
     # to the console that remains visible after exit
     CreateShortCut \
         "$InstDir\${LAUNCHER_SHORTCUT_NAME} Debug.lnk" \
-        "%COMSPEC%" '/K "$PythonExecPrefix\python.exe" -m "{LAUNCHERMODULE} -l4' \
+        "%COMSPEC%" '/K "$PythonExecPrefix\python.exe" "-m ${LAUNCHERMODULE}" -l4' \
```
Comment (ales-erjavec) [inline scripts/windows/orange-conda.nsi]: Note that this whole `/K "$PythonExecPrefix\python.exe" "-m ${LAUNCHERMODULE}" -l4` will be recorded verbatim as the shortcut parameters. Therefore the quotes must not group both `-m` and `${LAUNCHERMODULE}` into a single argument. In  fact neither require it as they are by definition single tokens (they cannot contain white space).

File: scripts/windows/orange-conda.nsi [hunk 4]
```diff
@@ -666,7 +666,7 @@ Section -Register SectionRegister
         "$PythonPrefix\share\${ICONDIR}\OrangeOWS.ico"
     WriteRegStr SHELL_CONTEXT \
         "Software\Classes\${INSTALL_REGISTRY_KEY}\Shell\Open\Command\" "" \
-        '"$PythonExecPrefix\pythonw.exe" -m "{LAUNCHERMODULE} "%1"'
+        '"$PythonExecPrefix\pythonw.exe" "-m ${LAUNCHERMODULE}" "%1"'
```
Comment (ales-erjavec) [inline scripts/windows/orange-conda.nsi]: Same as above

File: scripts/windows/orange-conda.nsi (hunk 2 above)
Comment (ales-erjavec) [reply scripts/windows/orange-conda.nsi]: No. The quoting here should be avoided.

File: scripts/windows/orange-conda.nsi (hunk 3 above)
Comment (astaric) [reply scripts/windows/orange-conda.nsi]: I see, the quotes are around the third parameter to the CreateShortCut. Fixed.

**Pull request merged**